design/providers/kubernetes_secrets_storage_csi/conjur_csi_provider_solution_design.md

```diff
@@ -0,0 +1,314 @@
+
+## Backwards compatibility
+
+The solution is implemented as a new provider for the Secrets Store CSI Driver and will not impact existing providers or Kubernetes workloads not using this new provider.
+
+## Performance
+
+Performance of this solution depends on the responsiveness of the CyberArk Conjur instance and the network latency between the Kubernetes cluster and the Conjur instance. The CyberArk Conjur provider itself introduces minimal overhead as it simply passes secrets from Conjur to the Secrets Store CSI driver.
+
+## Affected Components
+
+- **New Component:** CyberArk Conjur provider for Secrets Store CSI driver.
+- **Existing Component:** Secrets Store CSI driver.
+
+## Security
+
+The solution upholds high standards of security:
+
+1. Communication between the CyberArk Conjur provider and Conjur happens over a secure channel.
+2. `authn-jwt` ensures strong authentication with the use of cryptographically verifiable service account tokens.
+
+## Test Plan
+
+A comprehensive testing strategy will be employed:
+
+1. **Unit tests:** Test individual functions and methods in the CyberArk Conjur provider.
+2. **Integration tests:** Test the interaction of the CyberArk Conjur provider with the Secrets Store CSI driver and Conjur.
+3. **End-to-end tests:** Test the entire workflow from a Kubernetes workload to fetching secrets from Conjur.
+
+## Logs
+
+Both the Secrets Store CSI driver and the CyberArk Conjur provider will generate logs. These will provide insights into authentication, secret retrieval, and error conditions. We only have explicit control of the provider logs.
+
+## Documentation
+
+The documentation will cover:
+
+1. The deployment and configuration of the CyberArk Conjur provider.
+2. The creation and usage of a `SecretProviderClass` that utilizes the CyberArk Conjur provider.
+3. The configuration of Kubernetes workloads to use the CyberArk Conjur provider.
+
+## Open questions
+
+- Does the mount path have to be `/mnt/secrets-store` ?
+- What's the best way to configure things like CA certs for the provider etc. ? Perhaps allowing the `SecretProviderClass` to reference other Kubernetes resources such `ConfigMap` or `Secret` is the answer
+- What are our options for supporting authentication via authenticators other than authn-jwt ?
+- What is the upgrade strategy for the CyberArk Conjur provider deployed via Helm?
+- How to handle secret rotation in CyberArk Conjur?
```

Comment:
Looks like the [CSI Driver has alpha support for secret rotation](https://secrets-store-csi-driver.sigs.k8s.io/topics/secret-auto-rotation.html)

Reply:
Thanks, updated this